test/luatest_helpers/vtest.lua

```diff
@@ -535,10 +543,39 @@ local function router_new(g, name, cfg)
     g[name] = server
     g.cluster:add_server(server)
     server:start()
+    return server
+end
+
+--
+-- Create a new router in the cluster.
+--
+local function router_new(g, name, cfg)
+    local server = server_build_and_start(g, name)
     router_cfg(server, cfg)
     return server
 end
 
+--
+-- Create a new router inside fiber. pre_func is used to
+-- declare function, which will be run before starting
+-- router configuration. opts are options passed to pre_func.
+-- They are supposed to have the table type
+--
+local function router_fiber_new(g, name, cfg, pre_func, opts)
+    local server = server_build_and_start(g, name)
+    if pre_func then
+        server:exec(pre_func, opts)
+    end
+    router_fiber_cfg(server, cfg)
+    return server
+end
+
+local function router_fiber_new_join(router)
+    router:exec(function()
+        if f then f:join() end
+    end)
+end
+
```

Comment:
You might want to run luacheck locally. I don't remember how to install it - something about `luarocks` or `tarantoolctl rocks`. Anyway, I installed 100 years ago, run it it like that: `../tarantool/.rocks/bin/luacheck --codes --config .luacheckrc .` (I have `.rock` dir in my tarantool dir, which I have at the same level as vshard dir). Your commit produces these warnings:
```
Gerold103Macbook:vshard gerold$ ../tarantool/.rocks/bin/luacheck --codes --config .luacheckrc .
Checking test/luatest_helpers/vtest.lua           4 warnings

    test/luatest_helpers/vtest.lua:530:9: (W111) setting non-standard global variable f
    test/luatest_helpers/vtest.lua:531:9: (W113) accessing undefined variable f
    test/luatest_helpers/vtest.lua:575:12: (W113) accessing undefined variable f
    test/luatest_helpers/vtest.lua:575:19: (W113) accessing undefined variable f

Checking test/router-luatest/router_test.lua      15 warnings

    test/router-luatest/router_test.lua:454:20: (W112) mutating non-standard global variable ivshard
    test/router-luatest/router_test.lua:457:9: (W111) setting non-standard global variable old_cfg
    test/router-luatest/router_test.lua:458:40: (W113) accessing undefined variable old_cfg
    test/router-luatest/router_test.lua:462:13: (W111) setting non-standard global variable static_router
    test/router-luatest/router_test.lua:463:26: (W113) accessing undefined variable static_router
    test/router-luatest/router_test.lua:463:46: (W113) accessing undefined variable static_router
    test/router-luatest/router_test.lua:470:19: (W113) accessing undefined variable old_cfg
    test/router-luatest/router_test.lua:471:9: (W111) setting non-standard global variable old_info
    test/router-luatest/router_test.lua:477:30: (W113) accessing undefined variable static_router
    test/router-luatest/router_test.lua:477:52: (W113) accessing undefined variable static_router
    test/router-luatest/router_test.lua:488:20: (W113) accessing undefined variable old_info
    test/router-luatest/router_test.lua:489:47: (W113) accessing undefined variable fiber
    test/router-luatest/router_test.lua:498:9: (W112) mutating non-standard global variable ivshard
    test/router-luatest/router_test.lua:507:28: (W113) accessing undefined variable static_router
    test/router-luatest/router_test.lua:512:28: (W113) accessing undefined variable static_router
```

Reply:
My bad, I've done that but only for sources. I stand corrected)